### PR TITLE
fix playlist plugin for guest mode

### LINKED ISSUE #19794: 🐛 Bug Report: Catalog is unusable as a guest user
### 📜 Description

I'm unable to test anything as a guest user in my local environment due to a propogated error message that I have to click through to see the Entity page. I was able to do this locally as expected on branches from last month, but am no longer able to. Logging in through Github OAuth causes the issue to disappear.

https://github.com/backstage/backstage/assets/34432188/314432dd-92ab-47ba-bc79-d1c01baac8dc

I think this was introduced by https://github.com/backstage/backstage/pull/19434/files#diff-b49f50e782910892a50ac2d8092e3c1aa9498c9569da8574c3555f81a92b07b7R79 as the stack trace has that as the line causing it,
```
eval @ webpack-internal:///../core-app-api/src/apis/implementations/FetchApi/IdentityAuthInjectorFetchMiddleware.ts:43
eval @ webpack-internal:///../core-app-api/src/apis/implementations/FetchApi/PluginProtocolResolverFetchMiddleware.ts:40
getAllPlaylists @ webpack-internal:///../../plugins/playlist/src/api/PlaylistClient.ts:57
PlaylistEditDialog @ webpack-internal:///../../plugins/playlist/src/components/PlaylistEditDialog/PlaylistEditDialog.tsx:82
renderWithHooks @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:14985
updateFunctionComponent @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:17356
beginWork @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:19063
beginWork$1 @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:23935
performUnitOfWork @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:22771
workLoopSync @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:22702
renderRootSync @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:22665
performSyncWorkOnRoot @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:22288
eval @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:11327
unstable_runWithPriority @ webpack-internal:///../../node_modules/react-dom/node_modules/scheduler/cjs/scheduler.development.js:468
runWithPriority$1 @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:11276
flushSyncCallbackQueueImpl @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:11322
flushSyncCallbackQueue @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:11309
discreteUpdates$1 @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:22415
discreteUpdates @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:3756
dispatchDiscreteEvent @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:5889
```

### 👍 Expected behavior

The page should either
1. Throw an alert when I am unauthorized for the playlist plugin.
2. Now load playlists until I open the playlist modal.

### 👎 Actual Behavior with Screenshots

Every entity page has an error boundary like so and the alert box telling me I'm unauthorized.

Screen Shot 2023-09-05 at 5 07 59 PM

Network response is
```
{"error":{"name":"NotAllowedError","message":"Unauthorized","stack":"NotAllowedError: Unauthorized\n    at evaluateRequestPermission (webpack-internal:///../../plugins/playlist-backend/src/service/router.ts:65:19)\n    at async eval (webpack-internal:///../../plugins/playlist-backend/src/service/router.ts:94:36)"},"request":{"method":"GET","url":"/"},"response":{"statusCode":403}}
```

Network stack trace is
```
eval @ webpack-internal:///../core-app-api/src/apis/implementations/FetchApi/IdentityAuthInjectorFetchMiddleware.ts:43
eval @ webpack-internal:///../core-app-api/src/apis/implementations/FetchApi/PluginProtocolResolverFetchMiddleware.ts:40
getAllPlaylists @ webpack-internal:///../../plugins/playlist/src/api/PlaylistClient.ts:57
PlaylistEditDialog @ webpack-internal:///../../plugins/playlist/src/components/PlaylistEditDialog/PlaylistEditDialog.tsx:82
renderWithHooks @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:14985
updateFunctionComponent @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:17356
beginWork @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:19063
beginWork$1 @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:23935
performUnitOfWork @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:22771
workLoopSync @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:22702
renderRootSync @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:22665
performSyncWorkOnRoot @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:22288
eval @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:11327
unstable_runWithPriority @ webpack-internal:///../../node_modules/react-dom/node_modules/scheduler/cjs/scheduler.development.js:468
runWithPriority$1 @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:11276
flushSyncCallbackQueueImpl @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:11322
flushSyncCallbackQueue @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:11309
discreteUpdates$1 @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:22415
discreteUpdates @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:3756
dispatchDiscreteEvent @ webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:5889
```

### 👟 Reproduction steps

1. Log in as the guest user for a local Backstage environment.
2. Navigate to any entity page.
3. See the errors that follow.

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by freben**
Also avoids reloading the playlists on each render.

Fixes #19794

### CODE REVIEW

**Review by sennyeya — APPROVED** [review]
Thanks for picking this up!

**Review by kuangp — APPROVED** [review]

File: packages/app-next-example-plugin/package.json [hunk 1]
```diff
@@ -33,7 +33,6 @@
     "postpack": "backstage-cli package postpack",
     "clean": "backstage-cli package clean"
   },
-  "sideEffects": false,
```
[the remark this answers is not in the document]
Comment (freben) [reply packages/app-next-example-plugin/package.json:36]: Sorry, that seems to be an unclean master and yeah it's due to #19830

File: packages/app-next-example-plugin/package.json [hunk 2]
```diff
@@ -33,7 +33,6 @@
     "postpack": "backstage-cli package postpack",
     "clean": "backstage-cli package clean"
   },
```
Comment (Rugvip) [inline packages/app-next-example-plugin/package.json:35]:
```suggestion
  },
  "sideEffects": false,
```

:D

**Review by Rugvip — APPROVED** [review]
🎉

**Pull request merged**